### ISSUE TITLE
2.23.0 broke the build with a browser only extension

### ISSUE THREAD

**Issue by billti** [microsoft/vscode-vsce#929]
See https://github.com/microsoft/qsharp/pull/1091 where I had to pin to 2.22.0 to get our build to work again.

With 2.23.0 I get an error message running "vsce package --pre-release" that 

```
 ERROR  Extension entrypoint(s) missing. Make sure these files exist and aren't ignored by '.vscodeignore':
  extension/out/extension.js
```

But the extension is in ./out/extension.js and always has been. You can see this specified at https://github.com/microsoft/qsharp/blob/main/vscode/package.json#L20 .

Pinning back to 2.22.0 resolves the issue.

I haven't had a chance to dig into this more yet. Raising ASAP for awareness.

**Comment by mweichert** [comment]
Having this issue as well.

**Comment by Kubessandra** [comment]
Same

**Comment by remcohaszing** [comment]
This also affects non-browser extensions.

**Comment by sandy081** [comment]
Looks like this is something do with yarn workspaces feature support - https://github.com/microsoft/vscode-vsce/issues/300

I believe those who are seeing this issue has your projects using yarn workspaces?

**Comment by sandy081** [comment]
Can you please try adding following flag `--yarn` and see if it fixes?

**Comment by goto40** [comment]
> your projects using yarn workspaces?

No, but npm workspaces... I build the vsix in a project embedded in an npm workspace. I have no problem with a previous version of vsce.

**Comment by remcohaszing** [comment]
I ran into this in [mdx-analyzer](https://github.com/mdx-js/mdx-analyzer), which uses npm workspaces.

The output of the failed GitHub action is available on https://github.com/mdx-js/mdx-analyzer/actions/runs/7756736096/job/21154662574

**Comment by lars-reimann** [comment]
> Looks like this is something do with yarn workspaces feature support - #300
> 
> I believe those who are seeing this issue has your projects using yarn workspaces?

NPM workspaces for me as well.

**Comment by billti** [comment]
Seems we have a common pattern. Our VS Code package is also part of an npm workspace (https://github.com/microsoft/qsharp/blob/main/package.json).

**Comment by sandy081** [comment]
I can reproduce this with npm workspaces. Will work on the fix. Thanks for info.

**Comment by sandy081** [comment]
workaround until the fix comes is to use `--yarn` flag.

**Comment by isidorn** [comment]
Thank you for reporting this and for helping us find the issue.
We have started the investigation and I expect us to have a fix next week.
Until then I suggest to use the older version of vsce `2.22.0`. 

Sorry about that and thank you for your patience 🙏

**Comment by cipherdmg** [comment]
Hello Team,

First VSCode is simply awesome.

I also hit this exact same issue since I am not using yarn workspaces but I am using npm workspaces.

I did notice though just the presence of a .yarn folder even though I don't use yarn or have yarn dependencies then it works even though it does not use yarn.  If I remove that folder then it breaks.

The paths to the extension are not resolving.

The npm.js getNpmDependencies() normalizes files from the directory where the package.json has workspaces in it which is 2 levels higher so I end up with normalized paths like this.

extension/../../apps/vscode/dist/extension.js

My package.json for my extension has "main": "./dist/extension.js", I also tried "main": "dist/extension.js",

So when it goes into package.js

    onFile(file) {
        this.entryPoints.delete(util.normalize(file.path));
        return Promise.resolve(file);
    }
    
    This code that tries to delete the entryPoints LaunchEntryPointProcessor tries to delete the entryPoints but since they don't match it does not get deleted and then onFile() is called and throws an error.
    
![vsce-v2 22 0-vs-v2 23 0](https://github.com/microsoft/vscode-vsce/assets/50668978/a5b6348d-45ed-456b-a7c4-e87fa38b29a6)

Hope the screen shot helps.   My monorepo is broken so I have to move back to v2.22.0 until this gets fixed.

Thanks
CipherDMG

**Comment by sedrubal** [comment]
I ran into an issue which looks exactly similar to this one. However, I don't use workspaces. The reason was, that I've installed the Ubuntu snap nodejs package, [which](https://github.com/nodejs/snap/issues/13) [is](https://github.com/nodejs/node/issues/37982) [broken](https://github.com/nodejs/node/issues/32430). ➡️ Don't use snap

**Comment by TylerLeonhardt** [comment]
@joaomoreno can we get some verification steps?

**Comment by cipherdmg** [comment]
Thanks team, this resolved my issue also.

**Comment by goto40** [comment]
Works for me, too. Thank you!
